### PR TITLE
Improve logs for backends owned by the Observability Traces and Profiling squad

### PR DESCRIPTION
**Pull request by fabrizio-grafana**
Audit and improve logs for backends owned by the Observability Traces and Profiling squad.

Todos:
- [x] Decide whether we want to also log queries or not

Fixes https://github.com/grafana/grafana/issues/76406

### CODE REVIEW

File: pkg/tsdb/grafana-pyroscope-datasource/service.go [hunk 1]
```diff
@@ -55,50 +60,71 @@ func newInstanceSettings(httpClientProvider httpclient.Provider, ac accesscontro
...
 }
 
 func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	logger.Debug("Successfully processed query request")
 	i, err := s.getInstance(ctx, req.PluginContext)
 	if err != nil {
+		logger.Debug("Successfully processed query request")
 		return err
 	}
+	logger.Debug("Successfully processed query request")
```
Comment (joey-grafana) [inline pkg/tsdb/grafana-pyroscope-datasource/service.go:83]: Why are we logging the same message? Can we update the error one on line 80. Also, could remove the one on line 77 and use line 83 to indicate success.

File: pkg/tsdb/grafana-pyroscope-datasource/service.go [hunk 2]
```diff
@@ -55,50 +60,71 @@ func newInstanceSettings(httpClientProvider httpclient.Provider, ac accesscontro
...
 	return i.CallResource(ctx, req, sender)
 }
 
 func (s *Service) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	logger.Debug("Successfully processed query request")
```
Comment (joey-grafana) [inline pkg/tsdb/grafana-pyroscope-datasource/service.go:88]: Similar comment to above!

File: pkg/tsdb/grafana-pyroscope-datasource/instance.go [hunk 3]
```diff
@@ -97,14 +102,17 @@ func (d *PyroscopeDatasource) profileTypes(ctx context.Context, req *backend.Cal
 func (d *PyroscopeDatasource) labelNames(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
 	res, err := d.client.LabelNames(ctx)
 	if err != nil {
+		logger.Error("Received error from client", "error", err)
```
Comment (joey-grafana) [inline pkg/tsdb/grafana-pyroscope-datasource/instance.go:105]: Would be great to also mention the method name in the error i.e. `LabelNames` in this case so we can separate it from other errors. Same in other places.

File: pkg/tsdb/grafana-pyroscope-datasource/service.go [hunk 4]
```diff
@@ -55,50 +58,110 @@ func newInstanceSettings(httpClientProvider httpclient.Provider, ac accesscontro
 }
 
 func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	loggerWithContext := s.logger.FromContext(ctx)
+	loggerWithContext.Debug("Processing queries", "queries", req.Queries)
```
Comment (joey-grafana) [inline pkg/tsdb/grafana-pyroscope-datasource/service.go:62]: Not certain here if there will be sensitive info in the logged query and someone without the proper rights will see the logged info. Would probably just remove logs with query to be safe.

File: pkg/tsdb/parca/resources.go [hunk 5]
```diff
@@ -56,74 +59,94 @@ func (d *ParcaDatasource) callProfileTypes(ctx context.Context, req *backend.Cal
...
 	res, err := d.client.Values(ctx, connect.NewRequest(&v1alpha1.ValuesRequest{LabelName: label[0]}))
 	if err != nil {
+		logger.Debug("callLabelValues errored", "error", err)
 		span.RecordError(err)
 		span.SetStatus(codes.Error, err.Error())
 		return err
 	}
 	data, err := json.Marshal(res.Msg.LabelValues)
 	if err != nil {
+		logger.Debug("callLabelValues errored", "error", err)
```
Comment (joey-grafana) [inline pkg/tsdb/parca/resources.go:137]: Please mention here that it errored in the marshal (and below that it errored in the send).

**Review by joey-grafana — APPROVED** [review]

File: pkg/tsdb/grafana-pyroscope-datasource/service.go (hunk 4 above)
Comment (aocenas) [reply pkg/tsdb/grafana-pyroscope-datasource/service.go]: hmm not sure about that, I mean I wouldn't log the full request as to not log some authentication strings but I think logging a query can be super useful for debugging.

**Review by adrapereira — CHANGES_REQUESTED** [review]
Needs a couple of small changes, but overall looks good!

**Review by adrapereira — APPROVED** [review]

**Pull request merged**